oauthenticator/oauth2.py

```diff
@@ -1,104 +1,135 @@
 """
-Base classes for Custom Authenticator to use GitHub OAuth with JupyterHub
```

Comment:
To avoid confusion / compatibility, can you put the new class in `oauthenticator/generic.py`, rather than changing what `oauthenticator.oauth2` provides and means?